src/app/groups/containers/user-group-invitations/user-group-invitations.component.ts

```diff
@@ -72,16 +79,32 @@ export class UserGroupInvitationsComponent implements OnDestroy {
     if (sortMeta) this.onFetch(sortMeta);
   }
 
-  onAccept(pendingRequest: PendingRequest): void {
+  openJoinGroupConfirmationDialog(pendingRequest: PendingRequest): void {
+    this.joinGroupConfirmationDialog = {
+      id: pendingRequest.group.id,
+      name: pendingRequest.group.name,
+      requireLockMembershipApprovalUntil: null,
+      requirePersonalInfoAccessApproval: 'none',
+      requireWatchApproval: false,
+    };
+  }
+
+  closeModal(): void {
+    this.joinGroupConfirmationDialog = undefined;
+  }
+
+  accept(confirmEvent: JoinGroupConfirmEvent): void {
```

Comment:
And so, if you have a `pendingJoinRequest` that you need to use to get the `id`, probably you should use the `approvals` from that variable as well. And so I am not sure you still need this `confirmEvent`. (it is not like the user could accept only a part of the approvals)

Reply:
Fixed